File: src/test/java/de/espend/idea/php/annotation/tests/inspection/AnnotationDeprecatedInspectionTest.java
package de.espend.idea.php.annotation.tests.inspection;

import de.espend.idea.php.annotation.inspection.AnnotationDeprecatedInspection;
import de.espend.idea.php.annotation.tests.AnnotationLightCodeInsightFixtureTestCase;

public class AnnotationDeprecatedInspectionTest extends AnnotationLightCodeInsightFixtureTestCase {
    public void setUp() throws Exception {
        super.setUp();
        myFixture.copyFileToProject("classes_deprecated.php");
    }

    public String getTestDataPath() {
        return "src/test/java/de/espend/idea/php/annotation/tests/inspection/fixtures";
    }

    public void testThatInspectionIsDisplayedForAnnotationClasses() {
        assertLocalInspectionContains("test.php", "<?php\n" +
                        "use Doctrine\\ORM\\Mapping as ORM;\n" +
                        "\n" +
                        "/**\n" +
                        " * @ORM\\E<caret>ntity()\n" +
                        " */\n" +
                        "class Foo\n" +
                        "{\n" +
                        "}",
                AnnotationDeprecatedInspection.MESSAGE
        );

        assertLocalInspectionContainsNotContains("test.php", "<?php\n" +
                        "use Doctrine\\ORM\\Mapping as ORM;\n" +
                        "\n" +
                        "/**\n" +
                        " * @ORM\\Foo<caret>bar()\n" +
                        " */\n" +
                        "class Foo\n" +
                        "{\n" +
                        "}",
                AnnotationDeprecatedInspection.MESSAGE
        );
    }
}

File: src/test/kotlin/de/espend/idea/php/annotation/tests/inspection/AnnotationDeprecatedInspectionTest.kt
package de.espend.idea.php.annotation.tests.inspection

import de.espend.idea.php.annotation.inspection.AnnotationDeprecatedInspection
import de.espend.idea.php.annotation.tests.AnnotationLightCodeInsightFixtureTestCase

class AnnotationDeprecatedInspectionTest : AnnotationLightCodeInsightFixtureTestCase() {
    public override fun setUp() {
        super.setUp()
        myFixture.copyFileToProject("classes_deprecated.php")
    }

    override fun getTestDataPath(): String {
        return "src/test/kotlin/de/espend/idea/php/annotation/tests/inspection/fixtures"
    }

    fun testThatInspectionIsDisplayedForAnnotationClasses() {
        assertLocalInspectionContains(
            "test.php",
            """
                <?php
                use Doctrine\ORM\Mapping as ORM;

                /**
                 * @ORM\E<caret>ntity()
                 */
                class Foo
                {
                }
            """.trimIndent(),
            AnnotationDeprecatedInspection.MESSAGE,
        )

        assertLocalInspectionContainsNotContains(
            "test.php",
            """
                <?php
                use Doctrine\ORM\Mapping as ORM;

                /**
                 * @ORM\Foo<caret>bar()
                 */
                class Foo
                {
                }
            """.trimIndent(),
            AnnotationDeprecatedInspection.MESSAGE,
        )
    }
}
